src/address.rs

```diff
@@ -75,7 +75,7 @@ impl fmt::Display for AddressError {
 }
 
 impl error::Error for AddressError {
-    fn cause(&self) -> Option<&error::Error> {
+    fn cause(&self) -> Option<&dyn error::Error> {
```

Comment:
This won't work until we bump MSRV I think.

Reply:
oh oops, I didn't mean that to be in this PR